src/lib/receiver.js

```diff
@@ -281,44 +289,48 @@ function createReceiver (opts) {
     }
 
     const fulfillment = cc.toFulfillmentUri(conditionPreimage)
-    debug('about to submit fulfillment: ' + fulfillment)
+
     // returning the promise is only so the result is picked up by the tests' emitAsync
-    return client.fulfillCondition(transfer.id, fulfillment)
+    return reviewPayment(transfer)
```

Comment:
You should catch sync errors and treat them the same way as async errors. I'd just do:

``` js
Promise.resolve(transfer)
  .then(reviewPayment)
  // same code after that
```